File: 0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def getIntersectionNode(self, A: ListNode, B: ListNode) -> Optional[ListNode]:
        a, b = A, B
        while a != b:
            if a: a = a.next
            else: a = B
            if b: b = b.next
            else: b = A
        return a

File: 0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode getIntersectionNode(ListNode A, ListNode B) {
        ListNode a = A; ListNode b = B;
        while(a != b) {
            a = a != null ? a.next : B;
            b = b != null ? b.next: A;
        }
        return a;
    }
}